Source/AppAuthTV/OIDTVTokenRequest.m

```diff
@@ -0,0 +1,154 @@
+                    authorizationCode:(nullable NSString *)code
+                          redirectURL:(nullable NSURL *)redirectURL
+                             clientID:(NSString *)clientID
+                         clientSecret:(nullable NSString *)clientSecret
+                                scope:(nullable NSString *)scope
+                         refreshToken:(nullable NSString *)refreshToken
+                         codeVerifier:(nullable NSString *)codeVerifier
+                 additionalParameters:
+                     (nullable NSDictionary<NSString *, NSString *> *)additionalParameters
+    OID_UNAVAILABLE_USE_INITIALIZER(@selector
+                                     (initWithConfiguration:
+                                                  deviceCode:
+                                                    clientID:
+                                                clientSecret:
+                                        additionalParameters:
+                                              ))
+
+- (instancetype)initWithConfiguration:(OIDTVServiceConfiguration *)configuration
+                           deviceCode:(NSString *)deviceCode
+                             clientID:(NSString *)clientID
+                         clientSecret:(NSString *)clientSecret
+                 additionalParameters:(NSDictionary<NSString *, NSString *> *)additionalParameters {
+  self = [super initWithConfiguration:configuration
+                            grantType:kOIDTVDeviceTokenGrantType
+                    authorizationCode:nil
+                          redirectURL:[[NSURL alloc] initWithString:@""]
+                             clientID:clientID
+                         clientSecret:clientSecret
+                                scope:nil
+                         refreshToken:nil
+                         codeVerifier:nil
+                 additionalParameters:additionalParameters];
+
+  if (self) {
+    _deviceCode = [deviceCode copy];
+  }
+  return self;
+}
+
+#pragma mark - NSCopying
+
+- (instancetype)copyWithZone:(nullable NSZone *)zone {
+  // The documentation for NSCopying specifically advises us to return a reference to the original
+  // instance in the case where instances are immutable (as ours is):
+  // "Implement NSCopying by retaining the original instead of creating a new copy when the class
+  // and its contents are immutable."
+  return self;
+}
+
+#pragma mark - NSSecureCoding
+
++ (BOOL)supportsSecureCoding {
+  return YES;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    NSString *deviceCode = [aDecoder decodeObjectOfClass:[NSString class] forKey:kDeviceCodeKey];
+    _deviceCode = deviceCode;
+  }
+  return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)aCoder {
+  [super encodeWithCoder:aCoder];
+  [aCoder encodeObject:_deviceCode forKey:kDeviceCodeKey];
+}
+
+- (OIDURLQueryComponent *)tokenRequestBody {
```

Comment:
Any reason not to call to super?

Reply:
I could not do it for tokenRequestBody since the method in the parent class is non-public